### PR TITLE
Feature/core 721 data set forms

### PR DESCRIPTION
**Pull request by Tschuck**

### CODE REVIEW

File: core/digital-twin-lib/src/DAppTwin.ts [hunk 1]
```diff
@@ -108,15 +110,17 @@ class DAppTwin extends DigitalTwin {
...
     // load all running dispatcher instances for containers
-    const [ descInstances, addFavInstances, removeFavInstances ] = await Promise.all([
+    const [descInstances, addFavInstances, removeFavInstances] = await Promise.all([
       dispatchers.descriptionDispatcher.getInstances(this.runtime),
       dispatchers.twinFavoriteAddDispatcher.getInstances(this.runtime),
       dispatchers.twinFavoriteRemoveDispatcher.getInstances(this.runtime),
     ]);
 
     // reset previous saving states
+    const beforeStates = this.dispatcherStates ? JSON.parse(JSON.stringify(this.dispatcherStates))
```
Comment (datrinh) [inline core/digital-twin-lib/src/DAppTwin.ts]: We should avoid JSON trick to deep clone. Use lodash if necessary :)

File: core/evancore.vue.libs/src/components/forms/control/control.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export default class ControlComponent extends mixins(EvanComponent) {
   /**
    * The value for the input field.
    */
-  @Prop() value: Array<any> | string;
+  @Prop() value: any;
```
Comment (datrinh) [inline core/evancore.vue.libs/src/components/forms/control/control.ts]: Why was this necessary?

File: core/evancore.vue.libs/src/components/forms/files/files.ts [hunk 3]
```diff
@@ -71,8 +71,8 @@ export default class FilesInputComponent extends mixins(ControlComponent) {
    * Empty text that shown, when no files are uploaded and the component is disabled.
    */
   @Prop({
-    default: [ ]
-  }) value: Array<any>;
+    default: [],
+  }) value: UIContainerFile[] | { files: UIContainerFile[] };
```
Comment (datrinh) [inline core/evancore.vue.libs/src/components/forms/files/files.ts]: Imo the Prop interface should be just `UIContainerFile[]` and the parent should handle passing the right format. This would also make working with it way easier in this component (see line 97).
Or would this be too much refactoring?

File: core/evancore.vue.libs/src/components/forms/files/files.vue [hunk 4]
```diff
@@ -18,87 +18,109 @@
...
-      v-else-if="value.length === 0"
-      v-html="$t(emptyText)">
+        v-html="$t(placeholder)"
+      />
     </div>
-    <div class="invalid-feedback" v-if="error">
+    <div
+      v-else-if="files.length === 0"
+      class="centered"
+      v-html="$t(emptyText)"
```
Comment (datrinh) [inline core/evancore.vue.libs/src/components/forms/files/files.vue]: Is this avoidable?

File: core/evancore.vue.libs/src/components/forms/json/json.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+  /**
+   * Current string formatted value of this control.
+   */
+  stringValue = '';
+
+  /**
+   * Is the current input wrong formatted?
+   */
+  invalidJSON = false;
```
Comment (datrinh) [inline core/evancore.vue.libs/src/components/forms/json/json.ts]:
```suggestion
  isInvalidJSON = false;
```
I'd love to have boolean flags with "is" prefix as a consensus :)

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue [hunk 6]
```diff
@@ -183,4 +183,17 @@ export default DigitalTwinDetailComponent;
   margin-left: auto;
   margin-right: auto;
 }
+
+/deep/ .content-card {
+  background: white;
+  border-radius: 4px;
+  width: 564px;
+  padding: 32px 24px;
```
Comment (datrinh) [inline dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue]:
```suggestion
  padding: 24px 24px;
```
new standard in the concept

File: dapps/digital-twin-detail/src/components/container/DataSetForm.ts [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+  /**
+   * Return the easy type definition from a ajv schema (e.g. used to detect file fields).
+   *
+   * @param      {any}      subSchema   ajv sub schema
+   * @return     {string}   The type.
+   */
+  static getSchemaType(subSchema: any): string {
+    // check if it's a file
+    if (subSchema && subSchema.$comment) {
```
Comment (datrinh) [inline dapps/digital-twin-detail/src/components/container/DataSetForm.ts]:
```suggestion
    if (subSchema?.$comment) {
```

File: dapps/digital-twin-detail/src/components/container/DataSetForm.ts [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+      let $comment;
+
+      try {
+        $comment = JSON.parse(subSchema.$comment);
+      } catch (ex) {
+        // could not parse comment
+      }
+
+      if ($comment && $comment.isEncryptedFile) {
```
Comment (datrinh) [inline dapps/digital-twin-detail/src/components/container/DataSetForm.ts]:
```suggestion
      if ($comment?.isEncryptedFile) {
```

File: core/evancore.vue.libs/src/components/forms/control/control.ts (hunk 2 above)
Comment (Tschuck) [reply core/evancore.vue.libs/src/components/forms/control/control.ts]: By adding the json form control, specially any value can be passed through the control it self and the derived component must decide, which values are allowed.
